crabs/detection_tracking/inference_model.py

```diff
@@ -0,0 +1,394 @@
+            Gaussian-blurred grayscaled input frame
+        """
+        # convert the frame to grayscale frame
+        gray_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+        # apply Gaussian blurring
+        blurred_frame = cv2.GaussianBlur(gray_frame, kernel_size, sigmax)
+
+        return gray_frame, blurred_frame
+
+    def compute_background_subtracted_frame(
+        self,
+        blurred_frame,
+        mean_blurred_frame,
+        max_abs_blurred_frame,
+    ):
+        """
+        Compute the background subtracted frame for the
+        input blurred frame, given the mean and max absolute frames of
+        its corresponding video.
+
+        Parameters
+        ----------
+        blurred_frame : np.array
+            Gaussian-blurred grayscaled input frame
+        mean_blurred_frame : np.array
+            mean of all blurred frames in the video
+        max_abs_blurred_frame : np.array
+            pixelwise max absolute value across all blurred frames in the video
+
+        Returns
+        -------
+        background_subtracted_frame : np.array
+            normalised difference between the blurred frame f and
+            the mean blurred frame
+        """
+        return (
+            ((blurred_frame - mean_blurred_frame) / max_abs_blurred_frame) + 1
+        ) / 2
+
+    def compute_motion_frame(
+        self,
+        frame_delta,
+        background_subtracted_frame,
+        mean_blurred_frame,
+        max_abs_blurred_frame,
+    ):
+        """
+        _summary_.
+
+        Parameters
+        ----------
+        frame_delta : int
+            difference in number of frames used to compute the motion
+            channel is computed
+        background_subtracted_frame : np.array
+            normalised difference between the blurred frame f and
+            the mean blurred frame
+        mean_blurred_frame : np.array
+            mean of all blurred frames in the video
+        max_abs_blurred_frame : np.array
+            pixelwise max absolute value across all blurred frames in the video
+
+        Returns
+        -------
+        motion_frame : np.array
+            absolute difference between the background subtracted frame f
+            and the background subtracted frame f+delta
+        """
+        # compute the blurred frame frame_idx+delta
+        _, blurred_frame_delta = self.apply_grayscale_and_blur(
+            frame_delta,
+            [5, 5],
+            0,
+        )
+        # compute the background subtracted for frame_idx + delta
+        background_subtracted_frame_delta = (
+            self.compute_background_subtracted_frame(
+                blurred_frame_delta,
+                mean_blurred_frame,
+                max_abs_blurred_frame,
+            )
+        )
+
+        # compute the motion channel for frame_idx
+        return np.abs(
+            background_subtracted_frame_delta - background_subtracted_frame,
+        )
+
+    def __inference(
```

Comment:
maybe rename to
```suggestion
    def run_inference_one_frame(
```
?